tests/test_metadata.py

```diff
@@ -0,0 +1,83 @@
+from mapreader import loader
+import pytest
+import os
+import pandas as pd
+import shutil
+
+@pytest.fixture
+def dir_name():
+    return "metadata_test_dir"
+
+#if there is extra info in the metadata
+def test_extra_metadata(dir_name):  
+    os.mkdir(dir_name)
+    files = ["file1.png", "file2.png"]
+    for file in files:
+        with open(f"{dir_name}/{file}", "w") as f:
+            pass
+    metadata_df = pd.DataFrame({"name":["file1.png", "file2.png", "file3.png"], "coord":[(1.1,1.5),(2.1,1.0),(3.1,4.5)], "other":["one","two","three"]})
+    metadata_df.to_csv(f"{dir_name}/metadata_df.csv", sep="|")
+    
+    my_files=loader(f"{dir_name}/*png")
+    assert len(my_files)==2
+    my_files.add_metadata(f"{dir_name}/metadata_df.csv")
```

Comment:
I disagree that this is the correct behaviour in the circumstance. I think that the default behaviour should be to warn the user if there is a mismatch between the directory contents and the metadata.csv file. 

So `images.py` should include logic equivalent to:
```
if len(loader(f"{dir_name}/*png")) != len(metadata_df):
    raise ValueError("mismatch between files and metadata")
```

And in `test_metadata.py`, there should be something like:
```
with pytest.raises(ValueError, match="mismatch"):
    my_files.add_metadata(f"{dir_name}/metadata_df.csv")
```

If there is a use case where the metadata _must_ mismatch the directory contents, then perhaps we could introduce an `ignore_errors` param, eg:
```
# An error is raised
with pytest.raises(ValueError, match="mismatch"):
    my_files.add_metadata(f"{dir_name}/metadata_df.csv")

#An error is not raised
my_files.add_metadata(f"{dir_name}/metadata_df.csv", ignore_errors=True)
```

Reply:
Am happy with this as a solution but definitely think adding an ignore_errors (maybe ignore_mismatch) should be an option.
Will update.